### ISSUE TITLE
IPV6 support

### ISSUE THREAD

**Issue by Trak-X** [darkk/redsocks#89]
I am using redsocks on android, and i could not find a way to make it works with IPV6.

I read the documentation, and i still cannot figure out how to set the config file so it can support connections coming from ip6tables. 

Is actually ipv6 support implemented? I could not find anything related, and i see that the last ticket asking this is also in pending (but that is from a few years ago).

Thank you,
Mike

**Comment by darkk** [comment]
IPv6 is not supported yet. What's your use-case for IPv6+Socks5 environment? Is it a real-world scenario or just a lab test?

**Comment by Trak-X** [comment]
For now i am testing a closed environment and if everything is right, i would like to release it with an app.
My story:

i have an api that sets proxy server on device using iptables - supports ipv4 and ipv6 proxy. i would like to start redsocks to listen for v4/v6 connections from iptables and sends them to a http proxy that knows 4v/v6 connections 

The proxy server is built in-house and we also have a domain that responds with v4/v6 ip's.

**Comment by darkk** [comment]
OK, but you should be aware, that `redsocks` is TCP proxy, so it'll just send raw IPv6 address in `CONNECT [f000:baaa::]:443 HTTP/1.1` request. Is it OK for you? Using redsocks as a HTTP proxy (`type = http-relay`) is bad idea due to current implementation.

Also, do you need to be able to connect to IPv6-based proxy? Do you have real clients with IPv6-only and dual-stack connectivity? Do you need [Happy Eyeballs](https://en.wikipedia.org/wiki/Happy_Eyeballs) to be implemented to support clients with broken IPv6 setup?

**Comment by Trak-X** [comment]
Yes. I am well aware and all dependencies are included. Our sysadmin has already verified your answer and he says that everything you enumerated is enabled and included in our project. We use Connect for all requests and we can receive http/1.0 and http/1.1

**Comment by lenormf** [comment]
Hi,

I'm also interested in IPv6 support, for a deployment in production. What would it take to support it for transparent/SOCKS4 proxying of TCP traffic?

Thanks.

**Comment by wongsyrone** [comment]
I wrote a software that used on OpenWrt router to redirect specific IPv4/IPv6 TCP traffic to it and relay to SOCKS5 server in noauth mode, it implements both buffer copy and zero copy methods. If someone has interest, he can check https://github.com/wongsyrone/transocks-wong
